src/dataaggregator/aggregator/policyPoller_test.go

```diff
@@ -74,48 +66,49 @@ var _ = Describe("PolicyPoller", func() {
 				clock.Increment(2 * TestPolicyPollerInterval * time.Second)
 				Eventually(database.RetrievePoliciesCallCount).Should(BeNumerically(">=", 2))
 			})
-		})
-		Context("when there is no policy", func() {
-			BeforeEach(func() {
-				database.RetrievePoliciesStub = func() ([]*PolicyJson, error) {
-					return []*PolicyJson{}, nil
-				}
-
-			})
-			It("no trigger generated", func() {
-				clock.Increment(2 * TestPolicyPollerInterval * time.Second)
-				Consistently(func() int {
-					lock.Lock()
-					defer lock.Unlock()
-					return len(triggerMap)
-				}).Should(Equal(0))
-			})
-		})
-		Context("when there are policies", func() {
-			BeforeEach(func() {
-				database.RetrievePoliciesStub = func() ([]*PolicyJson, error) {
-					return []*PolicyJson{&PolicyJson{AppId: testAppId1, PolicyStr: policyStr1}}, nil
-				}
 
-			})
-			It("triggers generated", func() {
-				clock.Increment(3 * TestPolicyPollerInterval * time.Second)
-				Eventually(func() int {
-					lock.Lock()
-					defer lock.Unlock()
-					return len(triggerMap)
-				}).Should(Equal(1))
+			Context("when retrieve policies and compute triggers successfully", func() {
+				var consumed chan map[string]*Trigger
+				BeforeEach(func() {
+					database.RetrievePoliciesStub = func() ([]*PolicyJson, error) {
+						return []*PolicyJson{&PolicyJson{AppId: testAppId1, PolicyStr: policyStr1}}, nil
+					}
+					consumed = make(chan map[string]*Trigger, 1)
+					consumer = func(policies []*PolicyJson, triggers map[string]*Trigger) {
+						consumed <- triggers
+					}
+				})
+				It("should call the consumer with the new triggers for every interval", func() {
+					clock.Increment(2 * TestPolicyPollerInterval * time.Second)
+					var triggerMap map[string]*Trigger
+					Eventually(consumed).Should(Receive(&triggerMap))
+					Expect(triggerMap[testAppId1]).To(Equal(&Trigger{
+						AppId: testAppId1,
+						TriggerRecord: &TriggerRecord{
+							InstanceMaxCount: 5,
+							InstanceMinCount: 1,
+							ScalingRules: []*ScalingRule{&ScalingRule{
+								MetricType:         "MemoryUsage",
+								StatWindowSecs:     300,
+								BreachDurationSecs: 300,
+								CoolDownSecs:       300,
+								Threshold:          30,
+								Operator:           "<",
+								Adjustment:         "-1",
+							}}},
+					}))
+				})
 			})
 		})
 	})
 	Context("Stop", func() {
 		BeforeEach(func() {
+			poller = NewPolicyPoller(logger, clock, TestPolicyPollerInterval, database, consumer)
 			poller.Start()
 			poller.Stop()
 		})
 
 		It("stops the polling", func() {
-			clock.Increment(5 * TestPolicyPollerInterval * time.Second)
```

Comment:
Why did you remove this? Without adjusting time, you haven't proven that Stop really works.